src/utils.rs

```diff
@@ -72,6 +74,29 @@ pub fn get_base_device_limits_from_adapter_limits(adapter_limits: &wgt::Limits)
     }
     wgt::Limits::downlevel_webgl2_defaults()
 }
+#[inline]
+pub fn get_base_device_limits_from_adapter_limits(adapter_limits: &wgt::Limits) -> wgt::Limits {
```

Comment:
That's better, it really felt like the base_device_limits fn should handle that in the first place.